partner_pricelist_search/models/res_partner.py

```diff
@@ -39,5 +53,6 @@ def filter_func(partner):
                 _("Pricelist field do not support search with the operator '%s'.")
                 % operator
             )
-        partners = self.with_context(prefetch_fields=False).search([])
+        domain = self.env.context.get("search_partner_domain", [])
+        partners = self.with_context(prefetch_fields=False).search(domain)
         return [("id", "in", partners.filtered(filter_func).ids)]
```

Comment:
Can you add this changes to simplify code:
https://github.com/OCA/partner-contact/pull/1350/commits/60571d4406f887cc731150deb4eb0ca95981dbf4#diff-d7fde32bb36763c53ba086144a1584254003162fa6c604c59b7dbe68d7c6767aL17-R19